packages/table/src/interactions/selectable.tsx

```diff
@@ -110,142 +124,92 @@ export class DragSelectable extends React.Component<IDragSelectableProps, {}> {
-        if (event.shiftKey && selectedRegions.length > 0) {
-            this.didExpandSelectionOnActivate = true;
-            onSelection(expandSelectedRegions(selectedRegions, region));
-            focusSelectionIndex = 0;
-        } else if (DragEvents.isAdditive(event) && this.props.allowMultipleSelection) {
-            onSelection(Regions.add(selectedRegions, region));
-            // the focus should be in the newly created region,
-            // which is at index of the current list of regions plus one,
-            // which is the length of the current list of regions
-            focusSelectionIndex = selectedRegions.length;
+        } else if (this.shouldExpandSelection(event)) {
+            this.handleExpandSelection(region);
+        } else if (this.shouldAddDisjointSelection(event)) {
+            this.handleAddDisjointSelection(region);
         } else {
-            onSelection([region]);
-            focusSelectionIndex = 0;
+            this.handleReplaceSelection(region);
         }
 
-        const fullFocusCellCoordinates: IFocusedCellCoordinates = {
-            col: focusCellCoordinates.col,
-            focusSelectionIndex,
-            row: focusCellCoordinates.row,
-        };
-        this.props.onFocus(fullFocusCellCoordinates);
-
         return true;
     }
 
     private handleDragMove = (event: MouseEvent, coords: ICoordinateData) => {
-        const nextSelectedRegions = this.getDragSelectedRegions(event, coords);
-        if (nextSelectedRegions == null) {
+        const {
+            allowMultipleSelection,
+            focusedCell,
+            locateClick,
+            locateDrag,
+            selectedRegions,
+            selectedRegionTransform,
+        } = this.props;
+
+        let region = allowMultipleSelection
+            ? locateDrag(event, coords, /* returnEndOnly? */ this.didExpandSelectionOnActivate)
+            : locateClick(event);
+
+        if (!Regions.isValid(region)) {
             return;
+        } else if (selectedRegionTransform != null) {
+            region = selectedRegionTransform(region, event, coords);
         }
+
+        const nextSelectedRegions = this.didExpandSelectionOnActivate
+            ? this.expandSelectedRegions(selectedRegions, region, focusedCell)
+            : Regions.update(selectedRegions, region);
+
         this.maybeInvokeSelectionCallback(nextSelectedRegions);
 
-        if (!this.props.allowMultipleSelection) {
-            // have the focused cell follow the selected region
-            const mostRecentRegion = nextSelectedRegions[nextSelectedRegions.length - 1];
-            const focusCellCoordinates = Regions.getFocusCellCoordinatesFromRegion(mostRecentRegion);
-            const fullFocusCellCoordinates: IFocusedCellCoordinates = {
-                col: focusCellCoordinates.col,
-                focusSelectionIndex: nextSelectedRegions.length - 1,
-                row: focusCellCoordinates.row,
-            };
```

Comment:
This pattern (line 193-197) is super common. This is what `FocusedCellUtils.toFullCoordinates` is for (used within the new `this. invokeOnFocusCallbackForRegion helper`.